### PR TITLE
fix: experimental svg types

### PR DESCRIPTION
**Pull request by ematipico**
## Changes

Closes https://github.com/withastro/astro/issues/12626

This PR fixes an issue where `experimental.svg` had incorrect types. 

The previous types allowed having an empty object `{}` but it failed with a boolean value, which how we document it.

This was reported on Discord: https://discord.com/channels/830184174198718474/1313850696126431263

## Testing

There was a failing test, which is now fixed

### CODE REVIEW

File: packages/astro/src/assets/utils/svg.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ export type SvgRenderMode = 'inline' | 'sprite';
 export function makeSvgComponent(
 	meta: ImageMetadata,
 	contents: Buffer | string,
-	options?: { mode?: SvgRenderMode },
+	options?: { mode: SvgRenderMode },
```
Comment (ematipico) [inline packages/astro/src/assets/utils/svg.ts:25]: Having an optional `mode` should be a bug. Users should either provide a boolean, or provide an object with explicit `mode`

File: packages/astro/src/core/config/schema.ts [hunk 2]
```diff
@@ -541,17 +540,19 @@ export const AstroConfigSchema = z.object({
...
 				])
 				.optional()
+				.default(ASTRO_CONFIG_DEFAULTS.experimental.svg)
 				.transform((svgConfig) => {
 					// Handle normalization of `experimental.svg` config boolean values
 					if (typeof svgConfig === 'boolean') {
-						return svgConfig ? ASTRO_CONFIG_DEFAULTS.experimental.svg : undefined;
+						return svgConfig
+							? {
+									mode: 'inline' as SvgRenderMode,
```
Comment (ematipico) [inline packages/astro/src/core/config/schema.ts:553]: This cast make TS happy

File: packages/astro/test/fixtures/core-image-svg/astro.config.mjs [hunk 3]
```diff
@@ -4,6 +4,6 @@ import { defineConfig } from 'astro/config';
 export default defineConfig({
   integrations: [mdx()],
 	experimental: {
-		svg: {}
```
Comment (ematipico) [inline packages/astro/test/fixtures/core-image-svg/astro.config.mjs:7]: This way of passing the option shouldn't be valid

Comment (bluwy) [reply packages/astro/test/fixtures/core-image-svg/astro.config.mjs:7]: IIUC passing `{}` is like passing `true`. Couldn't we support `{}` without specifying `mode` too? It feels to me futureproofing if it ever has more optional options in the object, so you don't have to specify `mode` always.

Comment (ematipico) [reply packages/astro/test/fixtures/core-image-svg/astro.config.mjs]: I applied the change in [`26c65f2` (#12625)](https://github.com/withastro/astro/pull/12625/commits/26c65f25c7700ad356b0c358296873f0080c2cec) and tested that `svg: {}` is still a valid configuration

Comment (bluwy) [reply packages/astro/test/fixtures/core-image-svg/astro.config.mjs]: It doesn't mean a lot currently, but it's futureproofing if we add more options to it in the future, so I think it's an ok tradeoff for now as long as we don't recommend people to enable with `{}` only.

**Review by matthewp — APPROVED** [review]

**Pull request merged**